src/components/policy/policy_external/src/policy_manager_impl.cc

```diff
@@ -467,94 +466,109 @@ bool PolicyManagerImpl::LoadPT(const std::string& file,
   std::shared_ptr<policy_table::Table> pt_update = Parse(pt_content);
   if (!pt_update) {
     LOG4CXX_WARN(logger_, "Parsed table pointer is NULL.");
-    update_status_manager_.OnWrongUpdateReceived();
-    return false;
+    return PtProcessingResult::kWrongPtReceived;
   }
 
   file_system::DeleteFile(file);
   auto current_vd_items = GetVehicleDataItems();
 
   FilterPolicyTable(pt_update->policy_table, current_vd_items);
   if (!IsPTValid(pt_update, policy_table::PT_UPDATE)) {
-    wrong_ptu_update_received_ = true;
-    update_status_manager_.OnWrongUpdateReceived();
-    return false;
+    LOG4CXX_WARN(logger_, "Received policy table update is not valid");
+    return PtProcessingResult::kWrongPtReceived;
   }
 
-  update_status_manager_.OnValidUpdateReceived();
   cache_->SaveUpdateRequired(false);
+  sync_primitives::AutoLock lock(apps_registration_lock_);
 
-  {
-    sync_primitives::AutoLock lock(apps_registration_lock_);
+  // Get current DB data, since it could be updated during awaiting of PTU
+  std::shared_ptr<policy_table::Table> policy_table_snapshot =
```

Comment:
auto

Reply:
@LuxoftAKutsan fixed in https://github.com/smartdevicelink/sdl_core/pull/3051/commits/cd1b519dd9acd9a5e3ad93a849aa3d98076556d3